lib/dpul_collections/indexing_pipeline/figgy/resource.ex

```diff
@@ -53,25 +53,83 @@ defmodule DpulCollections.IndexingPipeline.Figgy.Resource do
 
   def extract_related_data(resource) do
     %{
-      "member_ids" => extract_members(resource),
-      "parent_ids" => extract_parents(resource)
+      "parent_ids" => extract_parents(resource),
+      "resources" => fetch_related(resource)
     }
   end
 
-  @spec extract_members(resource :: %__MODULE__{}) :: related_resource_map()
-  defp extract_members(resource = %{:metadata => %{"member_ids" => _member_ids}}) do
-    # Enum.reduce(member_ids, %{}, &append_related_resource/2)
-    # get just the list of members
-    # turn it into a map of id => FiggyResource
-    IndexingPipeline.get_figgy_members(resource.id)
+  @doc """
+  Finds all metadata properties which contain references to related resources
+  (those with the form `[%{"id" => id}]` and then fetches those resources from Figgy
+  in a single query.
+
+  ## Example
+
+  ```
+  r = %Figgy.Resource{
+        id: "097263fb-5beb-407b-ab36-b468e0489792",
+        internal_resource: "EphemeraFolder",
+        metadata: %{
+          "genre": [%{"id" => "668a21d7-750d-477d-b569-54ad511f13d7"}],
+          "member_ids": [%{"id" => "557cc7c1-9852-471b-ae4d-f1c14be3890b"}]
+        }
+      }
+
+  fetch_related(r)
+
+  Returns:
+  %{
+    "668a21d7-750d-477d-b569-54ad511f13d7" => %{
+      "id" => "668a21d7-750d-477d-b569-54ad511f13d7",
+      "internal_resource" => "EphemeraTerm",
+      "metadata" => %{
+        "label" => ["a genre"]
+      }
+    },
+    "557cc7c1-9852-471b-ae4d-f1c14be3890b" => %{
+      "id" => "557cc7c1-9852-471b-ae4d-f1c14be3890b",
+      "internal_resource: "FileSet",
+      "metadata" => %{
+        "file_metadata" => [
+          %{
+            "id" => %{"id" => "0cff895a-01ea-4895-9c3d-a8c6eaab4017"},
+            "internal_resource" => "FileMetadata",
+            "mime_type" => ["image/tiff"],
+            "use" => [%{"@id" => "http://pcdm.org/use#ServiceFile"}]
+          }
+        ]
+      }
+    }
+  """
+  @spec fetch_related(%__MODULE__{}) :: map()
```

Comment:
Suggestion: Non-blocking, but there's a `related_data()` type in this file, maybe that can be used as the return value in this typespec.

Reply:
Good idea. Fixed.